pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -250,16 +254,24 @@ def _warn_if_devices_flag_ignored(self) -> None:
         if self.devices is None:
             return
         devices_warning = f"The flag `devices={self.devices}` will be ignored, as you have set"
-        if self.distributed_backend in ("auto", _AcceleratorType.TPU):
+        if self.distributed_backend in ("auto", _AcceleratorType.TPU) or isinstance(
```

Comment:
what if we call `select_accelerator_type` first and then call `_warn_if_devices_flag_ignored`, then we can just use `self.use_ipu/cpu/gpu/tpu` here? will it have any side-effects?

Reply:
yeah, we can't as we call `_map_devices_to_accelerator` in `select_accelerator_type`, to assign gpus, ipus, tpu_cores when devices are passed to Trainer.